plugins/defaultplugins/aclplugin/vppdump/dump_vppcalls.go

```diff
@@ -122,25 +281,124 @@ func DumpMacIPAcl(log logging.Logger, vppChannel *govppapi.Channel, timeLog meas
 		}
 	}()
 
+	aclMACIPRules := make(map[ACLIdentifier][]acl_api.MacipACLRule)
+	var wasErr error
+
 	req := &acl_api.MacipACLDump{}
 	req.ACLIndex = 0xffffffff
 	reqContext := vppChannel.SendMultiRequest(req)
 	for {
-		msg := &acl_api.MacipACLDump{}
+		msg := &acl_api.MacipACLDetails{}
 		stop, err := reqContext.ReceiveReply(msg)
 		if err != nil {
-			return err
+			log.Error(err)
+			wasErr = err
 		}
 		if stop {
 			break
 		}
-		log.Info(msg.ACLIndex)
+
+		identifier := ACLIdentifier{
+			ACLIndex: msg.ACLIndex,
+			Tag:      string(bytes.Trim(msg.Tag, "\x00")),
+		}
+
+		aclMACIPRules[identifier] = msg.R
 	}
-	return nil
+	return aclMACIPRules, wasErr
 }
 
-// getIPACLDetails gets details for a given IP ACL from VPP and translates
-// them from the binary VPP API format into the ACL Plugin's NB format.
+// DumpInterfaceAcls finds interface in VPP and returns its ACL configuration
+func DumpInterfaceAcls(log logging.Logger, swIndex uint32, vppChannel *govppapi.Channel,
+	timeLog measure.StopWatchEntry) (acl.AccessLists, uint8, error) {
+
+	log.Info("DumpInterfaceAcls")
+	// ACLInterfaceListDump time measurement
+	alAcls := acl.AccessLists{
+		Acl: []*acl.AccessLists_Acl{},
+	}
+
+	start := time.Now()
+	defer func() {
+		if timeLog != nil {
+			timeLog.LogTimeEntry(time.Since(start))
+		}
+	}()
+
+	res, err := vppcalls.DumpInterface(swIndex, vppChannel, timeLog)
+	log.Infof("Res: %+v\n", res)
+	if err != nil {
+		return alAcls, 0, err
+	}
+
+	if res.SwIfIndex != swIndex {
+		return alAcls, 0, fmt.Errorf(fmt.Sprintf("Returned interface index %d does not match request",
+			res.SwIfIndex))
+	}
+
+	for aidx := range res.Acls {
+		ipACL, err := getIPACLDetails(vppChannel, uint32(aidx))
+		if err != nil {
+			log.Error(err)
+		} else {
+			alAcls.Acl = append(alAcls.Acl, ipACL)
+		}
+	}
+	return alAcls, res.NInput, nil
+}
+
+func getIPRuleDetails(rule acl_api.ACLRule) (*acl.AccessLists_Acl_Rule, error) {
+	// Resolve rule actions
+	actions := &acl.AccessLists_Acl_Rule_Actions{}
+	switch rule.IsPermit {
```

Comment:
This switch seems to be redundant for IPRule.. and MacIPRule.., should be extracted to separate function.